docs/design-proposals/node/Unified-Vertical-Scaling-for-Container-and-VM-Pods.md

```diff
@@ -319,3 +341,129 @@ Other components:
+
+    1. Numa support
+    2. Less memory fragmentation
+    3. Better control from VM spec patching
+
+2. VM memory live update
+
+    Both attach and detach device operations are done via the AttachDeviceFlag and DetachDeviceFlag API, with flag to
+    affect the running VM and persist the changes to the domain configuration for next reboot as well.
+       
+       AttachDeviceFlags(memoryDeviceDefinition, libvirt.DOMAIN_DEVICE_MODIFY_CONFIG|libvirt.DOMAIN_DEVICE_MODIFY_LIVE)
+       DetachDeviceFlags(memoryDeviceDefinition, libvirt.DOMAIN_DEVICE_MODIFY_CONFIG|libvirt.DOMAIN_DEVICE_MODIFY_LIVE)
+   
+3. Handle the async operation for memory unplug
+
+   Add a handler for the VIR_DOMAIN_EVENT_ID_DEVICE_REMOVED (future works). Current design returns in the UpdateContainerResource()
+   once after the DetachDeviceFlag() returns success without checking if the memory device is removed at the VM configuration in
+   the guest. It is done such way mainly for not holding the UpdateContainerResource() API for unexpected longer time due to
+   memory unplug.
+   
+4. Some limitations/constraints for VM workload types
+
+    1. Memory device size: 
+       Currently it is single size, with 128MiB each. i.e. if a VM patch increases the memory as 1Gi, 8 memory devices 
+       will be attached to the VM. The operations will still in one UpdateContainerResource() call though. i.e. Arktos VM 
+       runtime service will handle the attach operation internally with the calculation of memory increment in the patch.
+       
+    2. Max memory scale-up value
+       Currently a VM memory slots are set to 16 in Arktos VM runtime, so 16 * 128 Mi, which is 2Gi max scale-up value.
+       This is assumed the host has plenty allocatable memory.
+       
+    3. Guest OS Kernel requirements
+       Guest OS kernel must support device hotPlug and hotUnplug.
+       
+    4. HotUnPlug failures
+       Some times the OS just cannot migrated the app and free up the memory pages for the memory device. This can cause
+       memory unplug failures. lower the device size can mitigate but still not guaranteed. The Kubelet will retry in this case
+       till it successfully unplugged.
+
+#### API changes
+  As described in the "CRI changes" section, the Arttos-vm-runtime UpdateContainerResource() were implemented to update
+  the VM CGroup settings and update the VM domain definition with the new resource configuration.
+
+  If Kubelet enables its CPU manager and set the cpus to a particular pod. the cpuset is sent to the runtime via the 
+  UpdateContainerResource() and the runtime will honer the cpusets in both VM definition and its CGroup.
+
+  Two new APIs, SetVcpus() and SetCurrentMemory() were added to support VM vertical scaling, which extends the call to 
+  Libvirt APIs to update the VM domain xml.
+
+	// Update vcpu for a give domain
+	SetVcpus(int) error
+	// Update current memory for a given domain
```

Comment:
I leave a comment in code PR that suggests renaming the function to "ChangeXXX", "UpdateXXX" or similar names. The current name implies the input parameter is the new amount, but actually it's the delta.

Reply:
consinder renaming the api names

Reply:
Updated doc.